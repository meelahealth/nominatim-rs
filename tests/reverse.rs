#[tokio::test]
async fn reverse() {
    let client = nominatim::Client::new(
        url::Url::parse("https://nominatim.openstreetmap.org/").unwrap(),
        "nominatim-rust/0.1.0 test-suite".to_string(),
        Some("john_t@mailo.com".to_string()),
    )
    .unwrap();

    let reverse = client
        .reverse(
            nominatim::ReverseQueryBuilder::default()
                .address_details(true)
                .lon(13.53918)
                .lat(52.5460941)
                .zoom(nominatim::Zoom::Building)
                .build()
                .unwrap(),
        )
        .await
        .unwrap();

    println!("{}", serde_json::to_string_pretty(&reverse).unwrap());

    let reverse_expected: nominatim::Response =
        serde_json::from_str(include_str!("./reverse_expected.json")).unwrap();

    assert_eq!(reverse, reverse_expected);
}
