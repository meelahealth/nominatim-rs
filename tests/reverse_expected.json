{
    "place_id": 182143190,
    "licence": "Data Â© OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
    "osm_type": "way",
    "osm_id": 391407797,
    "lat": "52.54479867088309",
    "lon": "13.539087085722539",
    "display_name": "Wiesenburger Weg, Gewerbepark Marzahn, Marzahn, Marzahn-Hellersdorf, Berlin, 12681, Deutschland",
    "address": {
        "road": "Wiesenburger Weg",
        "industrial": "Gewerbepark Marzahn",
        "suburb": "Marzahn",
        "borough": "Marzahn-Hellersdorf",
        "city": "Berlin",
        "state": "Berlin",
        "postcode": "12681",
        "country": "Deutschland",
        "country_code": "de"
    },
    "boundingbox": ["52.5447731", "52.5449556", "13.5369099", "13.5394436"]
}
